### ISSUE TITLE
Results without AWSConv？

### ISSUE THREAD

**Issue by Alien1007** [joe-siyuan-qiao/DetectoRS#23]
I have found that the SAC is implemented based on AWSConv, which is not mentioned in ur DetectoRS paper. Have you tested standard nn.Conv2d in DetectoRS and what's the difference? What's more, it seems that AWSConv is not mentioned in Weight Standardization paper, is it similar to scaling and shifting in BN?

**Comment by joe-siyuan-qiao** [comment]
Hi, thanks for the question.

AWSConv keeps the mean and std of the weights loaded from the pre-trained checkpoints on ImageNet. So no matter what the gradients they receive when fine-tuning on COCO, they cannot get too far away from the pre-trained checkpoints. They are like adding anchors to the weights.

We used it because when playing with pre-trained backbones, it was very easy to get a NaN loss. Fixing the statistics of weights expands the explore space of the architecture design as it's almost unlikely to get a NaN loss as long as we don't push it too hard. As for the performance impact, it depends. It probably will hurt the final performance when training large models for a very long time as fixing statistics reduces the effective solution space. But without it, sometimes we got a NaN in the middle and wasted plenty of time. So it's highly recommended to have them.

**Comment by Alien1007** [comment]
Thanks for your reply. I have replaced nn.Conv2d with SAConv2d in BasicBlock of my model. Then I use nn.Module.load_state_dict(model_weights) to load a pre-trained model. However, I encounter many missing keys and unexpected keys. Do you have any idea?

**Comment by joe-siyuan-qiao** [comment]
I've never tried basicblocks myself as many mmdetection features are not supported for them. But if you are seeing unexpected keys other than those in the final fullly-connected layer, it's likely that you are loading a pre-trained model that does not match the model. For the missing keys, if they are something like weight_gamma or weight_beta, you can ignore them. They'll be computed during loading. Other missing keys will imply an error somewhere.

**Comment by Alien1007** [comment]
Yep, without replacing nn.Conv2d with SAConv2d, the model can be loaded correctly. And it seems that you have used SAC in Bottleneck rather than BasicBlock?

**Comment by joe-siyuan-qiao** [comment]
That's correct. We haven't done any experiments with BasicBlock.

**Comment by Alien1007** [comment]
Now I understand. Thanks a lot !
